libs/MeshKernel/benchmark/src/perf_curvilinear_uniform.cpp

```diff
@@ -62,3 +62,49 @@ BENCHMARK(BM_CurvilinearUniform)
     ->Args({2000, 2000})
     ->Args({4000, 4000})
     ->Args({5000, 5000});
+
+static void BM_CurvilinearUniform_add_faces_to_left_boundary(benchmark::State& state)
+{
+    for (auto _ : state)
+    {
+        // pause the timers to prepare the benchmark (excludes operation
+        // that are irrelevant to the benchmark and should not be measured)
+        state.PauseTiming();
+
+        const int numColumns = static_cast<int>(state.range(0));
+        const int numRows = static_cast<int>(state.range(1));
+        const double block_size = 10.0;
+
+        const double origin_x = 0.0;
+        const double origin_y = 0.0;
+        const double angle = 0.0;
+        const double blockSizeX = block_size;
+        const double blockSizeY = block_size;
+
+        CurvilinearGridCreateUniform const curvilinear_grid_create_uniform(Projection::cartesian);
+        const auto curvilinearGrid = std::make_shared<CurvilinearGrid>(
```

Comment:
Here a grid in created and computed, which involves instantiating and populating matrix  / vector of vectors. If you are not interested in this, pausing and resuming the timing can be removed all together.